src/hazelweb/Model.re

```diff
@@ -394,7 +403,28 @@ let perform_edit_action = (model: t, a: Action.t): t => {
   | Failed => raise(FailedAction)
   | CursorEscaped(_) => raise(CursorEscaped)
   | CantShift => raise(CantShift)
-  | Succeeded(new_edit_state) => model |> update_edit_state(new_edit_state)
+  | Succeeded(new_edit_state) =>
+    let new_model = model |> update_edit_state(new_edit_state);
+    let new_history = {
```

Comment:
instead of doing this match inline, lets create a function `UndoHistory.undoable_action : Action.t -> bool` and call that here